folding/utils/runandlog.py

```diff
@@ -0,0 +1,142 @@
+import subprocess
+import tqdm
+from typing import List, Dict
+from collections import defaultdict
+import bittensor as bt 
+import copy
+import wandb
+from loguru import logger
+from dataclasses import asdict, dataclass
+
+import folding
+
+
+"""
+We need something thet gets returned (result dict) 
+We don't want it to grow each time its called
+We want it to be independent of gromacs
+We dont need to be worried about pdb_id 
+This needs to log to the event dict 
+"""
+
+
+def output_dict():
+    return defaultdict(str)
+
+class CommandFailedException(Exception):
+    pass
+
+class RunAndLog():
```

Comment:
maybe it makes sense that there is a parent `Log` class, similar to what we had before but where it holds all the methods needed for wandb logging. Maybe we call it `WandbLogger`, and then we do `class RunAndLog(WandbLogger)`. This way we can just call the logger (or any other type of logger when we want it inside of the codebase. I see that you do this right now by trying to call `log_event`. This is a potential solution. And if we ever have to move away from Wandb, it also gives us the capabilities to write another Logger method as we see fit

Reply:
I added this approach in recent commits.